### PR TITLE
Inbound OML Support

### PR DESCRIPTION
**Pull request by JohnNKing**
This PR adds support for inbound OML (laboratory order) messages. These are similar to the ORM (generic order) messages that are already supported.

Test Steps:
1. Send a test OML message to the Waters API (e.g. https://github.com/CDCgov/prime-reportstream/blob/d453a322cdd9a06990dc9bef5c639474a919c491/prime-router/src/testIntegration/resources/datatests/HL7_to_FHIR/sample_oml_20240319-001.hl7)
2. Message will be received successfully (HTTP 201), whereas previously this would fail
3. The message should then be delivered to the CDC TI service

## Changes
- OML_021 messages are now allowed
- The Flexion / CDC-TI receiver now accepts both ORM and OML messages
- Tests have been added to cover inbound OML messages

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? **n/a** 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced? **n/a** 
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes? **n/a?**
- [ ] Database changes are submitted as a separate PR?  **n/a**
- [ ] DevOps team has been notified if PR requires ops support? **n/a** 

## Linked Issues
- Accomplishes https://github.com/CDCgov/trusted-intermediary/issues/962

## To Be Done
- https://github.com/CDCgov/trusted-intermediary/issues/819

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints? **No**
- Does this PR include changes in authentication and/or authorization of existing endpoints? **No**
- Does this change introduce new dependencies that need vetting? **No**
- Does this change require changes to our infrastructure? **No**
- Does logging contain sensitive data? **No**
- Does this PR include or remove any sensitive information itself? **No**

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/test/kotlin/fhirengine/translation/HL7toFhirTranslatorTests.kt [hunk 1]
```diff
@@ -68,7 +68,7 @@ OBX|1|CWE|94558-4^SARS-CoV-2 (COVID-19) Ag [Presence] in Respiratory specimen by
 
         // Source: https://confluence.hl7.org/display/OO/v2+Sample+Messages
         val unsupportedHL7 = """
-MSH|^~\&#|NIST EHR|NIST EHR Facility|NIST Test Lab APP|NIST Lab Facility|20130211184101-0500||OML^O21^OML_O21|NIST-LOI_5.0_1.1-NG|T|2.5.1|||AL|AL|||||
+MSH|^~\&#|NIST EHR|NIST EHR Facility|NIST Test Lab APP|NIST Lab Facility|20130211184101-0500||OML^O33^OML_O33|NIST-LOI_5.0_1.1-NG|T|2.5.1|||AL|AL|||||
```
Comment (JohnNKing) [inline prime-router/src/test/kotlin/fhirengine/translation/HL7toFhirTranslatorTests.kt:71]: Note: This is suppose to fail. Adding support for OML_O21 means I had to change this to an unsupported message type.

File: prime-router/metadata/test_fhir_mapping/hl7/message/OML_O21.yml
Comment (JohnNKing) [inline prime-router/metadata/test_fhir_mapping/hl7/message/OML_O21.yml:1]: I'm not actually using this test mapping at present... should I? Or should this change be removed?

File: frontend-react/src/styles/_components.scss [hunk 2]
```diff
@@ -115,4 +115,3 @@ hr {
         margin-top: 40px;
     }
 }
-
```
Comment (jpandersen87) [inline frontend-react/src/styles/_components.scss:118]: Undo this change to prevent chromatic from running.

Comment (JohnNKing) [reply frontend-react/src/styles/_components.scss]: Thank you @jpandersen87! Fixed. (I'm at a bit of a loss how this made it into the merge commit to begin with...)

**Review by MauriceReeves-usds — APPROVED** [review]
This all looks fine to me, but I would love if @oslynn @victor-chaparro or @arnejduranovic could also review.

File: prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt [hunk 3]
```diff
@@ -126,7 +127,9 @@ class HL7Reader(private val actionLogger: ActionLogger) : Logging {
                         v27_ORU_R01::class.java,
                         v251_ORU_R01::class.java
                     )
-
+                    "OML" -> listOf(
```
[the remark this answers is not in the document]
Comment (victor-chaparro) [reply prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt:130]: ORM should also be added. I just tried processing an ORM message and got the warning.

[the remark this answers is not in the document]
Comment (JohnNKing) [reply prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt:130]: @victor-chaparro Any objections to me making that part of a separate PR? I was going to add it, but attempting to do so is inexplicably causes unit tests to fail:

```
Failed to map supported failure 'org.opentest4j.AssertionFailedError: expected to be true' with mapper 'org.gradle.api.internal.tasks.testing.failure.mappers.OpenTestAssertionFailedMapper@17bc1586': Cannot invoke "Object.getClass()" because "obj" is null

> Task :test
gov.cdc.prime.router.fhirengine.utils.FHIRBundleHelpersTests Test enhance bundle metadata 2-7() FAILED

  org.opentest4j.AssertionFailedError: expected to be true
      at app//gov.cdc.prime.router.fhirengine.utils.FHIRBundleHelpersTests.Test enhance bundle metadata 2-7(FHIRBundleHelpersTests.kt:672)


FAILURE: Executed 1097 tests in 1m 36s (1 failed, 4 skipped)
```

File: prime-router/src/test/kotlin/fhirengine/translation/HL7toFhirTranslatorTests.kt (hunk 1 above)
Comment (oslynn) [reply prime-router/src/test/kotlin/fhirengine/translation/HL7toFhirTranslatorTests.kt:71]: Is this for fail test?

**Review by oslynn — APPROVED** [review]

**Comment by oslynn** [comment]
Looking good to me.

File: prime-router/src/test/kotlin/fhirengine/translation/HL7toFhirTranslatorTests.kt (hunk 1 above)
Comment (JohnNKing) [reply prime-router/src/test/kotlin/fhirengine/translation/HL7toFhirTranslatorTests.kt:71]: @oslynn -- I believe so, yes. There's a test for a valid ORU message before this -- then this message is used as part of an assertFailure.

**Review by arnejduranovic — APPROVED** [review]
